pkg/auth/dependency/mfa/authenticator.go

```diff
@@ -113,3 +114,60 @@ func MaskAuthenticators(authenticators []interface{}) []interface{} {
 	}
 	return output
 }
+
+func CanAddAuthenticator(authenticators []interface{}, newA interface{}, mfaConfiguration config.MFAConfiguration) bool {
+	// Calculate the count
+	totalCount := len(authenticators)
+	totpCount := 0
+	oobSMSCount := 0
+	oobEmailCount := 0
+	for _, a := range authenticators {
+		switch aa := a.(type) {
+		case TOTPAuthenticator:
+			totpCount++
+		case OOBAuthenticator:
+			switch aa.Channel {
+			case coreAuth.AuthenticatorOOBChannelSMS:
+				oobSMSCount++
+			case coreAuth.AuthenticatorOOBChannelEmail:
+				oobSMSCount++
+			default:
+				panic("unknown OOB authenticator channel")
+			}
+		default:
+			panic("unknown authenticator")
+		}
+	}
+
+	// Simulate the count if new one is added.
+	totalCount++
```

Comment:
Can we extract this switch as a local function? The dupe code in the loop doesn't look good...